api/src/pcapi/core/offerers/api.py

```diff
@@ -1936,6 +1936,9 @@ def search_bank_account(search_query: str, *_: typing.Any) -> BaseQuery:
     else:
         filters.append(finance_models.BankAccount.id == dehumanized_id)
 
+    if search_query.isnumeric():
```

Comment:
Pour éviter de laisser passer avec des caractères comme `²`, qui causeront une exception dans `int()`, tu peux utiliser `string_utils.is_numeric` : 
```suggestion
    if string_utils.is_numeric(search_query):
```